brave-tests/src/test/java/brave/internal/recorder/PendingSpansClassLoaderTest.java

```diff
@@ -60,18 +61,16 @@
       pendingSpans.getOrCreate(context, true);
       context = null; // orphan the context
 
+      System.gc();
       try {
-        blockOnGC();
+        // We usually block on a weak reference being cleared, but here we would end up loading
+        // the WeakReference class from the classloader preventing it from unloading.
```

Comment:
non-blocking comment: is the WeakReference class the problem here? or is it the type it holds

Reply:
Pretty sure it's the `WeakReference` class itself - it's a bit of conjecture on my part, though, not exactly sure why trying to block on weak reference never works here.